python/observation-publisher/publisher/prediction_log_consumer.py

```diff
@@ -141,9 +142,54 @@ def __init__(
 
         self._consumer = Consumer(consumer_config)
         self._batch_size = config.batch_size
+        self._topic = config.topic
         self._consumer.subscribe([config.topic])
         self._poll_timeout = config.poll_timeout_seconds
 
+        background_job_thread = Thread(target=self._emit_metrics)
+        background_job_thread.setDaemon(True)
+        background_job_thread.start()
+
+    def _emit_metrics(self):
```

Comment:
I want to confirm something, is the objective of this function is to calculate the kafka lag every 60 seconds?

Reply:
yes

Reply:
> please reconfirm what is the acceptable interval

Are we going to remove this comment in https://github.com/caraml-dev/merlin/pull/598/files#diff-95854776df8c83183573e19f3bf6e2bfe61f73ccc4077d92ee9b64385c13d403R159 then? 😅

Reply:
when will this function be called repeatedly to log the lag, afaik the `start()` on `Thread` only call the target function once?

Reply:
nice catch! it should be looping forever in target function